pkg/db/src/implementations/version-1.1/note.ts

```diff
@@ -211,18 +211,13 @@ class Note implements NoteInterface {
 	}
 
 	/**
-	 * Add volumes accepts a tuple update (isbn, quantity, warehouse) or an array of tuples (updates) for
+	 * Add volumes accepts an object or multiple objects of VolumeStockOptionalWarehouse updates (isbn, quantity, warehouse?) for
 	 * book quantities. If a volume with a given isbn is found, the quantity is aggregated, otherwise a new
 	 * entry is pushed to the list of entries.
 	 */
 	addVolumes(...params: Parameters<NoteInterface['addVolumes']>): Promise<NoteInterface> {
-		// A check to see if the update function should be ran once or multiple times
-		const isTuple = (params: Parameters<NoteInterface['addVolumes']>): params is VolumeTransactionTuple => {
-			return !Array.isArray(params[0]);
-		};
-
 		return runAfterCondition(async () => {
-			const updateQuantity = (isbn: string, quantity: number, warehouseId = this.#w._id) => {
+			const updateQuantity = (isbn: string, quantity: number, warehouseId: `v${number}/${string}`) => {
 				const matchIndex = this.entries.findIndex((entry) => entry.isbn === isbn && entry.warehouseId === warehouseId);
```

Comment:
This is ok, only you have `VersionedString` type for this usage (alias for `${number}/${string}`)

Reply:
Also, since we don't need to check for `isTuple`, we don't need to declare an `updateQuantity` function, could just run its body inside the top level function.